### PR TITLE
DEV-1710 consider unit testing migrations

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### What's this PR do?

- Updates Python tests in CI to require migration names to be non-automatic.
- Updates name of one migration in users app as required by above
- Updates README with a short section on migration requirements and best practices

#### Why are we doing this? How does it help us?

Make migrations easier to understand and maintain

#### How should this be manually tested?

N/A

#### Have automated unit tests been added?

N/A

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

N/A

#### Has this been documented? If so, where?

Yes, in README.

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-1710 | Consider unit testing migrations](https://news-revenue-hub.atlassian.net/browse/DEV-1710)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them?

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -434,6 +437,21 @@ REACT_APP_HUB_V3_GOOGLE_ANALYTICS_ID=therealvaluehere
...
+
+When you're working on a new feature, we have some minimal guidelines around Django migrations that you are expected to follow.
+
+### No automatic migration names
+
+First, be sure to give your migrations a meaningful, non-automatic name. By default, Django will often give a name like `0003_auto_20220801_1100.py`. Let's imagine that migration removes a column ('my_column') from a model ('MyModel'). A better name would be `0003_DEV-1234_drop_my_column_from_mymodel.py`, which references the ticket name and describes at a high level what the migration does. This practice helps maintainers have a high level understanding of what a migration does simply by reading its file name. To this end, there is a Make command (`make test_migrations`) that runs as part of the Python tests in CI which prohibits automatic migration names.
+
+### Squash PR-related migrations
+
+We want to limit the number of individual migration files produced. As you're working on a feature branch, oftentimes you'll iteratively create numerous migrations for a given app as you run down feature implementation.
```
Comment (x110dc) [inline README.md:450]: Maybe include a link here to https://docs.djangoproject.com/en/4.0/topics/migrations/#migration-squashing?

File: revengine/settings/base.py [hunk 2]
```diff
@@ -453,3 +454,9 @@
 
 # Add reversion models to admin interface
 ADD_REVERSION_ADMIN = True
+
+#
+DTM_IGNORED_MIGRATIONS = {
```
Comment (x110dc) [inline revengine/settings/base.py:459]: Maybe a comment about what DTM means and what this is here for? django-test-migrations I assume?

**Review by x110dc — CHANGES_REQUESTED** [review]
Minor requests.

File: revengine/settings/base.py (hunk 2 above)
Comment (benjaminEwhite) [reply revengine/settings/base.py:461]: Added explanatory comment

File: README.md (hunk 1 above)
Comment (benjaminEwhite) [reply README.md:450]: Adding link

**Review by x110dc — APPROVED** [review]

**Pull request merged**